src/frontend/http.rs

```diff
@@ -878,9 +896,16 @@ pub mod tests {
         #[case] maybe_headers: Option<Vec<(&'_ str, &'_ str)>>,
         #[values(None, Some("test_db"))] new_db: Option<&str>,
         #[values("", ".bin")] extension: &str,
+        #[values(None, Some("max-age=3600, public"))] cache_control: Option<&str>,
```

Comment:
Come to think about it, are these rstest cases multiplicative? i.e. if you add two values for cache-control here, do we end up running twice as many tests?

Reply:
Yeah it's essentially a Cartesian product of the `case` and `values` sets, so we end up with all the previous tests run twice now.

```bash
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_1_None::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_1_None::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_1_None::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_1_None::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_2_Some___test_db___::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_2_Some___test_db___::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_2_Some___test_db___::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_2_Some___test_db___::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_1_None::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_1_None::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_1_None::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_1_None::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_2_Some___test_db___::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_2_Some___test_db___::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_2_Some___test_db___::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_2_Some___test_db___::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_1_None::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_1_None::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_1_None::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_1_None::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_2_Some___test_db___::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_2_Some___test_db___::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_2_Some___test_db___::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_2_Some___test_db___::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_1_None::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_1_None::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_1_None::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_1_None::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_2_Some___test_db___::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_2_Some___test_db___::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_2_Some___test_db___::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_2_Some___test_db___::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
```